doc/harpoon.txt

```diff
@@ -0,0 +1,123 @@
+harpoon.txt*    For Neovim version 0.5                 Last change: 2022 Nov 16
+
+===============================================================================
+CONTENTS                                                     *harpoon-contents*
+
+  Introduction ········································· |harpoon-introduction|
+  Requirements ········································· |harpoon-requirements|
+  Usage ······················································· |harpoon-usage|
+    Commands ··············································· |harpoon-commands|
+    Lua API ················································· |harpoon-lua-api|
+  Configuration ·············································· |harpoon-config|
+  Highlights ············································· |harpoon-highlights|
+  License ··················································· |harpoon-license|
+
+===============================================================================
+INTRODUCTION                                                          *harpoon*
+
+The goal of Harpoon is to get you where you want with the fewest keystrokes.
+
+The Problems:
+    * You're working on a codebase. medium, large, tiny, whatever. You find
+    yourself frequenting a small set of files and you are tired of using a
+    fuzzy finder, :bnext & :bprev are getting too repetitive, alternate file
+    doesn't quite cut it, etc etc.
+    * You want to execute some project specific commands or have any number of
+    persistent terminals that can be easily navigated to.
+
+The Solutions:
+    * The ability to specify, or on the fly, mark and create persisting key
```

Comment:
This is very non-technical on given functions below. Typically, this includes a high-level overview of the core functionality (persistent JSON files for absolute file paths with marks, shell and tmux commands + interface, (limited) scratch buffer as user interface and lua functions to operate with those).